### PR TITLE
@mzikherman - Add artworks_connection to filter_artworks

### PR DESCRIPTION
**Pull request by broskoski**
Mostly PR-ing this as-is for discussion ( cc @alloy )

The one issue here is that we need the `total` count for results to make relay happy. However in the case of `filter_artworks`, this won't be present unless the query contains the `total` aggregation.

### CODE REVIEW

File: schema/filter_artworks.js [hunk 1]
```diff
@@ -24,6 +28,20 @@ export const FilterArtworksType = new GraphQLObjectType({
       description: 'Artwork results.',
       type: new GraphQLList(Artwork.type),
     },
+    artworks_connection: {
+      type: artworkConnection,
+      args: pageable(),
+      resolve: ({ hits, aggregations }, options) => {
+        if ((!aggregations && !aggregations.total)) {
+          throw new Error('This query must contain the total aggregation');
```
Comment (alloy) [inline schema/filter_artworks.js]: Why throw an error, can’t you just add the `total` field instead?

Comment (broskoski) [inline schema/filter_artworks.js]: The `total` field isn't available unless you specifically request the aggregation. For instance if you query ` counts { total } ` without specifying the `total` aggregation you will get an error also.

**Comment by broskoski** [comment]
What say you, @mzikherman ?

File: schema/filter_artworks.js (hunk 1 above)
Comment (alloy) [inline schema/filter_artworks.js]: I really have a hard time parsing this, isn’t this essentially doing the following?

```js
if (aggregations === null && aggregations.total === null) {
}
```

Which, when reduced more, equals:

```js
if (null && null.total === null) {
}
```

Which means the second conditional is never going to be evaluated.

Comment (broskoski) [inline schema/filter_artworks.js]: Right but `aggregations` can exist without `aggregations.total` existing.

```javascript
if ({ medium } && { medium }.total) 
```

Comment (broskoski) [inline schema/filter_artworks.js]: BTW, we don't want the second conditional to evaluate in your example!

Comment (mzikherman) [inline schema/filter_artworks.js]: Might the better conditional be: `if (!aggregations || !aggregations.total)`?

**Comment by mzikherman** [comment]
I like it! I'm for throwing the explicit `Error`. `count`s aren't free in ElasticSearch (and they are totally uncached in Mongo as we were discussing just yesterday!). 

Automatically adding query params, esp. ones that need to do 'work' makes me uneasy, seems like an easy way to accidentally DOS ourselves...

**Pull request merged**